### PR TITLE
WaddleParser tests

### PR DESCRIPTION
**Pull request by CFSY**

### CODE REVIEW

File: src/main/java/seedu/waddle/logic/parser/UnplanCommandParser.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 /**
  * Parses input arguments and creates a new UnplanCommand object
  */
-public class UnplanCommandParser {
+public class UnplanCommandParser implements Parser<UnplanCommand> {
```
Comment (csheun) [inline src/main/java/seedu/waddle/logic/parser/UnplanCommandParser.java:14]: Good catch, I think we missed that out for a few of the new classes.

File: src/main/java/seedu/waddle/logic/commands/CopyCommand.java [hunk 2]
```diff
@@ -39,4 +39,10 @@ public CommandResult execute(Model model) throws CommandException {
 
         return new CommandResult(String.format(MESSAGE_SUCCESS, selectedItinerary.getDescription()));
     }
+
+    @Override
+    public boolean equals(Object other) {
```
Comment (seox123) [inline src/main/java/seedu/waddle/logic/commands/CopyCommand.java:44]: Good catch adding the equal methods.

File: src/test/java/seedu/waddle/logic/parser/UnplanCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import static seedu.waddle.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.waddle.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.waddle.commons.core.index.Index;
+import seedu.waddle.commons.core.index.MultiIndex;
+import seedu.waddle.logic.commands.UnplanCommand;
+
+public class UnplanCommandParserTest {
```
Comment (csheun) [inline src/test/java/seedu/waddle/logic/parser/UnplanCommandParserTest.java:13]: LGTM, good job clement!

File: src/test/java/seedu/waddle/logic/parser/WaddleParserTest.java [hunk 4]
```diff
@@ -91,10 +132,23 @@ public void parseCommand_find() throws Exception {
         FindCommand command = (FindCommand) parser.parseCommand(
                 FindCommand.COMMAND_WORD + " " + keywords.stream().collect(Collectors.joining(" ")));
         assertEquals(new FindCommand(new NameContainsKeywordsPredicate(keywords)), command);
+
+        // switch to itinerary page
+        StageManager.getInstance().setWishStage(getGraduation());
+        assertThrows(ParseException.class, MESSAGE_UNAVAILABLE_COMMAND_ITINERARY, ()
+                -> parser.parseCommand(FindCommand.COMMAND_WORD
+                + " " + keywords.stream().collect(Collectors.joining(" "))));
```
Comment (csheun) [inline src/test/java/seedu/waddle/logic/parser/WaddleParserTest.java:140]: Impressive work!

File: src/test/java/seedu/waddle/testutil/TypicalItems.java [hunk 5]
```diff
@@ -73,7 +74,7 @@ public static Item getTour() {
 
     public static Item getBreakfast() {
         return new ItemBuilder().withDesc(VALID_ITEM_DESC_BREAKFAST)
-                .withDuration(VALID_DURATION_BREAKFAST).withCost(VALID_COST_ART)
+                .withDuration(VALID_DURATION_BREAKFAST).withCost(VALID_COST_BREAKFAST)
```
Comment (csheun) [inline src/test/java/seedu/waddle/testutil/TypicalItems.java:77]: Good catch!

**Review by csheun — APPROVED** [review]
LGTM, Keep up the good work!

File: src/test/java/seedu/waddle/logic/parser/WaddleParserTest.java [hunk 6]
```diff
@@ -123,6 +190,145 @@ public void parseCommand_unknownCommand_throwsParseException() {
         StageManager.getInstance().setHomeStage();
 
         assertThrows(ParseException.class, MESSAGE_UNKNOWN_COMMAND, ()
-            -> parser.parseCommand("unknownCommand"));
+                -> parser.parseCommand("unknownCommand"));
```
Comment (seox123) [inline src/test/java/seedu/waddle/logic/parser/WaddleParserTest.java:193]: Very important and extensive test cases, well done Clement!

File: src/test/java/seedu/waddle/logic/parser/UnplanCommandParserTest.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    private final UnplanCommandParser parser = new UnplanCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "     ",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, UnplanCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
```
Comment (seox123) [inline src/test/java/seedu/waddle/logic/parser/UnplanCommandParserTest.java:23]: Good test cases. LGTM.

**Review by seox123 — APPROVED** [review]
Good work Clement!

**Pull request merged**